### PR TITLE
implemented stringToDouble feature

### PR DESCRIPTION
**Pull request by velievrauf**
I suggest to add stringToDouble feature that extends the existing functional

### CODE REVIEW

File: cstring.h [hunk 1]
```diff
@@ -363,6 +363,40 @@ int64_t stringFindFirstCharArr(TString s, const char *pattern) {
     return -1; 
 }
 
+double stringToDouble(TString s) {
+    double result = 0.0;
+    int sign = 1;
+    int decimalIndex = -1;
+    if (s.size > 0 && s.data[0] == '-') {
+        sign = -1;
+        stringRemove(&s, 0, 1);
```
Comment (Tnirpps) [inline cstring.h:372]: Deleting the first element takes O(n), it's too much
It'll be better just set some sort of start_position to 1 and then start loop not from 0 but from start_position

File: cstring.h [hunk 2]
```diff
@@ -363,6 +363,40 @@ int64_t stringFindFirstCharArr(TString s, const char *pattern) {
...
+            break;
+        }
+    }
+    for (size_t i = 0; i < s.size; ++i) {
+        if (s.data[i] == '.') {
+            break;
+        }
+        if (stringCharIsDigit(s.data[i])) {
+            result = result * 10.0 + (s.data[i] - '0');
+        }
```
Comment (Tnirpps) [inline cstring.h:386]: `Else` (when it's not a digit) set error like "Not a Number" using `setError()`

About error handling [here](https://github.com/Tnirpps/cstring/pull/5#discussion_r1605905205)

File: cstring.h [hunk 3]
```diff
@@ -363,6 +363,40 @@ int64_t stringFindFirstCharArr(TString s, const char *pattern) {
...
+    }
+    double divisor = 10.0;
+    if (decimalIndex != -1) {
+        for (size_t i = decimalIndex + 1; i < s.size; ++i) {
+            if (stringCharIsDigit(s.data[i])) {
+                result += (s.data[i] - '0') / divisor;
+                divisor *= 10.0;
+            }
```
Comment (Tnirpps) [inline cstring.h:394]: Also `setError()`

File: tests/main.c [hunk 4]
```diff
@@ -372,6 +372,22 @@ void test_stringContains() {
     printGreen("test_stringContains\n");
 }
 
+void test_stringToDouble() {
+    TString s1 = stringInitWithCharArr("3.14159");
+    TString s2 = stringInitWithCharArr("123");
+    TString s3 = stringInitWithCharArr("-100");
+
```
Comment (Tnirpps) [inline tests/main.c:379]: Add test with empty string or invalid double `-12pas7 8.14`

P.S. you forgot to call this test in `main()`

**Review by Tnirpps — CHANGES_REQUESTED** [review]
